### PR TITLE
Bump gradle version and use detach() everywhere

### PR DESCRIPTION
**Pull request by joetimmins**
The previous PR https://github.com/novoda/landing-strip/pull/25 fixed an issue with the new `detach` method. So this PR makes every activity in the demo app actually make use of it.

Also bumps the gradle version so that we stay up to date with cool stuff like Instant Run!

### CODE REVIEW

**Review by niamh-power — APPROVED** [review]

File: demo/src/main/java/com/novoda/landingstrip/CustomLandingStrip.java [hunk 1]
```diff
@@ -8,17 +8,24 @@
...
         ViewPager viewPager = (ViewPager) findViewById(R.id.view_pager);
         viewPager.setAdapter(new DemoFragmentPagerAdapter(getSupportFragmentManager()));
 
         landingStrip.attach(viewPager, viewPager.getAdapter());
     }
 
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        landingStrip.detach();
```
Comment (zegnus) [inline demo/src/main/java/com/novoda/landingstrip/CustomLandingStrip.java]: I recall agreeing on doing `super.onCreate() ... your code` and `your code... super.onDestroy()` could you invert it in this PR? :)

Comment (joetimmins) [reply demo/src/main/java/com/novoda/landingstrip/CustomLandingStrip.java]: Can do mate 👍

Comment (joetimmins) [reply demo/src/main/java/com/novoda/landingstrip/CustomLandingStrip.java]: Fixed in https://github.com/novoda/landing-strip/pull/26/commits/35c3263475ace6ee535c2fe1b602d8f5ea5b849b

**Review by zegnus — APPROVED** [review]
